crypto/evp/evp_lib.c

```diff
@@ -148,6 +155,50 @@ int EVP_CIPHER_set_asn1_iv(EVP_CIPHER_CTX *c, ASN1_TYPE *type)
     }
     return i;
 }
+
+int evp_cipher_get_asn1_aead_params(EVP_CIPHER_CTX *c, ASN1_TYPE *type)
+{
+    int i = 0;
+    long tl;
+    unsigned char iv[EVP_MAX_IV_LENGTH];
+
+    if (type == NULL)
+        return i;
+
+    i = asn1_type_get_octetstring_int(type, &tl, NULL, EVP_MAX_IV_LENGTH);
+    if (i <= 0)
+        return -1;
+    asn1_type_get_octetstring_int(type, &tl, iv, i);
+
+    if (EVP_CIPHER_CTX_ctrl(c, EVP_CTRL_AEAD_SET_IVLEN, i, NULL) <= 0
+            || EVP_CIPHER_CTX_ctrl(c, EVP_CTRL_SET_IV, i, iv) <= 0)
+        return -1;
+
+    return i;
+}
+
+
+int evp_cipher_set_asn1_aead_params(EVP_CIPHER_CTX *c, ASN1_TYPE *type)
+{
+    int ivl, tl;
+    unsigned char iv[EVP_MAX_IV_LENGTH];
+
+    if (type == NULL)
+        return 0;
+
+    tl = EVP_CIPHER_CTX_ctrl(c, EVP_CTRL_AEAD_GET_TAGLEN, 0, NULL);
```

Comment:
@mattcaswell I have rebased the PR and started seeing issues likely caused by #9231 from @slontis as the GCM cipher is now moved to providers which currently returns `EVP_CTRL_RET_UNSUPPORTED` for this type. From the code, it looks that it's just a temporary thing (`default:` in the middle of switch and the providers params logic skipped) so I was wondering if I should just wait till this is addressed or I need to look for a different solution?

Reply:
I will look at this now - should be easy to fix..

Reply:
Are you supporting CCM also?

Reply:
See #9698

Reply:
Thanks for creating PR so quickly!

CCM is not currently supported by this PR as it's a bit more tricky when using BIO cipher and will require some further changes to get the input length which needs to be known in advance for CCM. Plan to look into it later.

Reply:
Yep the BIO's are not pretty :)

Reply:
This is no longer used so marking as resolved.